src/ConstantProductFactory.sol

```diff
@@ -49,6 +63,79 @@ contract ConstantProductFactory {
         settler = _settler;
     }
 
+    /**
+     * @notice Creates a new CoW AMM with the specified imput parameters.
+     * @param token0 The address of the first token in the pair.
+     * @param amount0 The initial amount of the first token in the pair.
+     * @param token1 The address of the second token in the pair.
+     * @param amount1 The initial amount of the second token in the pair.
+     * @param minTradedToken0 The minimum amount of token0 before the AMM
+     * attempts auto-rebalance.
+     * @param priceOracle The address of the price oracle to use for the AMM.
+     * @param priceOracleData The data to pass to the price oracle.
+     * @param appData The app data to pass to the AMM.
+     * @return amm The address of the newly deployed AMM.
+     */
+    function create(
+        IERC20 token0,
+        uint256 amount0,
+        IERC20 token1,
+        uint256 amount1,
+        uint256 minTradedToken0,
+        address priceOracle,
```

Comment:
nit: any particular reason this isn't typed with `IPriceOracle`?

Reply:
Fixed in 2194b9d7e568899b975a8e90fa1d1ebea68c3324.